### ISSUE TITLE
Reveal the source code, make it open-source with appropriate license

### ISSUE THREAD

**Issue by kutsan** [marta-file-manager/marta-issues#194]
Please, consider making it open-source. It will make it more trustable and contributors friendly.

You should also consider making it [free software](https://www.gnu.org/philosophy/free-sw.en.html).

**Comment by yanex** [comment]
Hello, Kutsan!

I thought a lot about this, and I came to the mind that open-sourcing Marta doesn't make any benefits both to me and Marta users.

1. I will have to maintain the open-source community. This includes code reviewing, some kind of code-related support, and I afraid I don't have enough time to do it properly.
2. Even if Marta were an open-source project, I suppose most of the contributions would be about new actions and custom file system support. But you actually don't need to change Marta source code for that. There is a public API available, and one can share plugins under the license he/she likes.
3. Despite the fact that making revenue from selling Marta is not my primary goal, I've spent significant time developing it, and it would be nice if I get some compensation. My experience says me selling open-source software is hard, and making money from donations is not really possible.
4. Trust is a huge point, but I never saw people who check the app source code before using it. There are lots of closed-source software for macOS and looks like nobody actually bothers about it.

Although this is my current mindset, it can change in the future.

But anyway, I want to express the strong commitment I have: 
- Marta will **never collect any user-related data** without the explicit agreement (and denying it won't cause any implications). No information is collected right now.
- There **always will be a perpetual license** (and it won't cost much). Until 1.0, Marta will be completely free.

**Comment by warpkanal** [comment]
Sounds meaningful to me, open-source as such doesn't automatically imply more/faster development. Checking a few open-source filemanagers actually shows the opposite, see e.g. DoubleCommander and mucommander (mostly one developer, abandonded over time, forked by a single other developer and so on).
I consider the most important thing in a filemanager's dev approach to be enduringness; there's a reason why TotalCommander reached its feature set, because it was steadily developed over decades :)
One interesting twist I found in the open vs. closed source debates is the statement from the author of fman: it's closed source and commercial, but he promised that should he ever not release a new version in some timeframe (don't recall the exact timeframe, maybe it was 6 months), he'll open-source the software.

**Comment by mishoboss** [comment]
I was so happy to find Marta two days ago. And now I'm so disappointed to read the above statement. Nobody does open source for MacOS and I really can't understand why...

**Comment by yanex** [comment]
@mishoboss Well, almost all high-quality open source projects are actually maintained (or sponsored) by commercial companies, and its developers are paid. I understand that the idea of making a better world is great, but few people would consider it if they have nothing to eat.

Desktop software doesn't bring much revenue (I don't talk here about professional stuff like Photoshop or IDEs), and there's little sense in sponsoring it, unless it's bloated with telemetry and viruses.

Look at the Linux world: the kernel itself is very stable, as lots of the maintainers are from good companies such as Intel or Red Hat. On the other hand, most of the desktop applications (DE, music players, calendars) are half-baked and buggy. I used Linux for many years until I migrated to Mac so I know that pretty well.

Open sourcing does not make software better for sure. And I remember a few cases when it led to the project death.

By the way, do you have some open source projects right now? :)

**Comment by mishoboss** [comment]
> By the way, do you have some open source projects right now? :)
I contributed several projects in the past and had one mine, but unfortunately I abandoned it - https://github.com/openhab/openhab1-addons/wiki/TouchUI

I know what you mean. Very well indeed. I just hate the Mac world where everyone is trying to make some buck for same stuff that in the Linux world it should be considered a shame for even thinking about that. It's strange how two UNIX based systems evolved into two very different developer's mentality directions... :))) MacOS software ecosystem has a lot of crap too, its not only Linux. Difference is that crap in Mac's ecosystem costs money. :) 

Anyways, Marta is great! Keep the good work!
